app/controllers/users_controller.rb

```diff
@@ -205,6 +205,72 @@ def check_emails
     render json: failed_json, status: 403
   end
 
+  def update_primary_email
+    if !SiteSetting.enable_secondary_emails
```

Comment:
Just to be 100% sure, when the site setting is disabled, people can still change their "primary" email address, right?

Reply:
Yes, they can. This endpoint is used just to switch between emails. To change the email address, `PUT /u/:username/preferences/email` is used instead.